imagetest/test_suites/security/image_security_test.go

```diff
@@ -0,0 +1,327 @@
+	}
+}
+
+func verifyPassword(image string) error {
+	fileBytes, err := ioutil.ReadFile("/etc/passwd")
+	if err != nil {
+		return err
+	}
+	users := strings.Split(string(fileBytes), "\n")
+	for _, user := range users {
+		// ignore empty user
+		if len(user) == 0 {
+			continue
+		}
+		passwdItems := strings.Split(user, ":")
+		loginname := passwdItems[0]
+		passwd := passwdItems[1]
+		uid := passwdItems[2]
+		shell := passwdItems[6]
+		// A passwd entry of length 1 is a disabled password.
+		if len(passwd) != 1 {
+			return fmt.Errorf("login for %s does not look disabled", loginname)
+		}
+
+		uidValue, err := strconv.Atoi(uid)
+		if err != nil {
+			return err
+		}
+		// Don't check user account and root account
+		if uidValue >= maxUID || uidValue == 0 {
+			continue
+		}
+		if targetShell, found := specialAccountShells[loginname]; found {
+			if targetShell != shell {
+				return fmt.Errorf("account %s has wrong login shell %s", loginname, shell)
+			}
+		} else {
+			// SUSE has bin user with login access
+			if !strings.Contains(image, "sles") && !strings.Contains(shell, "false") && !strings.Contains(shell, "nologin") {
+				return fmt.Errorf("account %s has the login shell %s", loginname, shell)
+			}
+		}
+	}
+	return nil
+}
+
+func verifySSHConfig() error {
+	fileBytes, err := ioutil.ReadFile("/etc/ssh/sshd_config")
+	if err != nil {
+		return err
+	}
+	sshdConfig := string(fileBytes)
+	if !strings.Contains(sshdConfig, "PasswordAuthentication no") {
+		return fmt.Errorf("\"PasswordAuthentication\" was not set to \"no\"")
+	}
+
+	noRootSSH := strings.Contains(sshdConfig, "PermitRootLogin no")
+	noRootPasswordSSH := strings.Contains(sshdConfig, "PermitRootLogin without-password")
+	if !noRootSSH && !noRootPasswordSSH {
+		return fmt.Errorf("\"PermitRootLogin\" was not set to \"no\" or \"without-password\"")
+	}
+	return nil
+}
+
+func verifySecurityUpgradeEnabled(image string) error {
+	var expecedBlock, expectedLine string
+	if strings.Contains(image, "debian") {
+		expecedBlock = unattendedUpgradeBlockDebian
+		expectedLine = expectedDebian
+	} else if strings.Contains(image, "ubuntu") {
+		expecedBlock = unattendedUpgradeBlockUbuntu
+		expectedLine = expectedUbuntu
+	}
+	bytes, err := ioutil.ReadFile(unattendedUpgradeConfigPath)
+	if err != nil {
+		return err
+	}
+	var inBlock bool
+	for _, line := range strings.Split(string(bytes), "\n") {
+		line = strings.TrimSpace(line)
+		switch {
+		case strings.Contains(line, "//"):
+			continue
+		case strings.Contains(line, expecedBlock):
+			inBlock = true
+		case inBlock && strings.Contains(line, expectedLine):
+			// Success!
+			return nil
+		case inBlock && strings.Contains(line, "};"):
+			return fmt.Errorf("incorrect Unattended-Upgrade config")
+		}
+	}
+	return fmt.Errorf("missing Unattended-Upgrade config")
+}
+
+func verifyUpgradesPackageInstalled() error {
+	if _, _, err := runCommand("dpkg-query", "-s", "unattended-upgrades"); err != nil {
+		return err
+	}
+	return nil
+}
```

Comment:
OK i was wrong on this one, debian doesn't make it so easy to determine if a package is correctly installed. here is suggested alternative:

```suggestion
func verifyDebianPackage(packageName string) error {
	stdout, _, err := runCommand("dpkg-query", "-W", "--showformat", "${Status}", packageName)
	if err != nil {
		return err
	}
	if strings.TrimSpace(stdout) != "install ok installed" {
	  return fmt.Errorf("package is not correctly installed")
	}
	return nil
}
```